File: Vectors , ArrayList, List/Python/MajorityElement[Leet_Code_169].py
# Find the element with majority count in the array.
# Majority element is the element that appears more than n/2 times in the array.
def majority(v):
    frequency = 0
    ans = 0
    for num in v:
        if frequency == 0:
            ans = num
        if ans == num:
            frequency += 1
        else:
            frequency -= 1
    return ans

if __name__ == "__main__":
    v = [1, 2, 2, 1, 1]
    majority_element = majority(v)
    print("Majority Element:", majority_element)

File: Vectors , ArrayList, List/C++/MajorityElement[Leet_Code_169].cpp
// Find the element with majority count in the array.
// Majority element is the element that appears more than n/2 times in the array.

#include <vector>
#include <iostream>
using namespace std;

int majority(vector<int>& v) {
    int frequency = 0, ans = 0;
    for (int i = 0; i < v.size();i++){
        if(frequency == 0)
            ans = v[i];
        if(ans == v[i])
            frequency++;
        else
            frequency--;    
    }
    return ans;
}

int main() {
    vector<int> v = {1, 2, 2, 1, 1};
    int majorityElement = majority(v);
    cout << "Majority Element: " << majorityElement << endl;
    return 0;
}
